### PR TITLE
Feature/upgrade explore api v2

### PR DESCRIPTION
**Pull request by remilejeune**
Updating the API documentation:
- Use Explore instead of Search
- highlight the V2
- rephrase the TPF explanation

### CODE REVIEW

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 1]
```diff
@@ -3,14 +3,9 @@ Using the API
 
 An API (Application Programming Interface) is an interface for programs to communicate and exchange data.
 
-.. admonition:: Note
-   :class: note
-
-   To know more about what APIs are and why and how they are used, you can read the article "`You don\'t need analogies to understand APIs <https://kitchen.opendatasoft.com/you-dont-need-analogies-to-understand-apis-14da4513f970>`_" written by one of our Opendatasoft developers.
```
Comment (baptabl) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:9]: Why removing the link to Julien's article? I find it very clear as a first jump into _what APIs are_.

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 2]
```diff
@@ -3,14 +3,9 @@ Using the API
...
-
 Opendatasoft provides access to 6 APIs:
 
-- `ODS Search API v1 <https://help.opendatasoft.com/apis/ods-search-v1/>`_ and `ODS Search API v2 <https://help.opendatasoft.com/apis/ods-search-v2/>`_: Opendatasoft's own APIs, used internally by the platform and the `ODS widgets <https://help.opendatasoft.com/widgets/>`_
+- `ODS Explore API V2 <https://help.opendatasoft.com/apis/ods-explore-v2/>`_: Opendatasoft's own APIs, used by the platform.
```
Comment (baptabl) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:8]: I think we should also highlight that it's our **main** public API here. Something like
```suggestion
- `ODS Explore API V2 <https://help.opendatasoft.com/apis/ods-explore-v2/>`_: main Opendatasoft explore API, used to explore catalogs and datasets with a custom SQL-like query language: ODSQL. Available publicly and also used internally.
```

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 3]
```diff
@@ -30,33 +25,31 @@ These APIs can be used, for example, to search for datasets and data, to compute
...
+ODS Explore API
+---------------
 
-This ODS Search API is used internally by the Opendatasoft platform and widgets. To develop an application, we recommend using this API.
+The ODS Explore API is used internally by the Opendatasoft platform and widgets. To develop an application, we recommend using this API, in V2.
 
-There are currently 2 versions of the ODS Search API: the `v1 <https://help.opendatasoft.com/apis/ods-search-v1/>`_ and the `v2 <https://help.opendatasoft.com/apis/ods-search-v2/>`_, which is still under development. Listed below, the main differences between the 2 versions.
+There are currently 2 versions of the ODS Explore API: the `V2 <https://help.opendatasoft.com/apis/ods-explore-v2/>`_ and the `V1 <https://help.opendatasoft.com/apis/ods-search-v1/>`_, which is now deprecated. Listed below, the main differences between the 2 versions.
 
-* The ODS Search API v2 introduces a better implementation of REST principles. Each API entry-point provides links to navigate between linked resources (HATEOAS) easily.
```
Comment (baptabl) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:40]: `a better implementation of REST principles` 🙈 -> `/query` 🎤 ⬇️

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 4]
```diff
@@ -30,33 +25,31 @@ These APIs can be used, for example, to search for datasets and data, to compute
...
-* Key changes in the v2:
+* Key changes in the V2:
 
   * A richer query language (ODSQL) is available, with new functions and arithmetic expressions.
-  * Special filters (such as geofilter and facet filters) are now directly integrated to ODSQL, in the ``where`` parameter (i.e ``geofilter.distance='42,1,100'`` is replaced by ``where=distance(geo_field,geom'Point(42 1)',100)``).
-  * The aggregation API (former analyze API) now shares the same structure and ODSQL language as the Search API. It is available on both the catalog (to aggregate datasets) and the dataset (to aggregate records) APIs.
-  * Exporters have their own entry-point under ``/api/catalog/exports`` and ``/api/catalog/datasets/<dataset_id>/exports``.
-  * Navigation search (former facet search) and standard search are completely separated.
+  * Special filters (such as geofilter and facet filters) are now directly integrated to ODSQL, in the ``where`` parameter.
+  * Aggregations endpoint and records endpoint now share the same structure and ODSQL language.
```
Comment (baptabl) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:39]: We agreed some time ago to deprecate the term `aggregations` which is very elastic-specific-language.
```suggestion
  * We deprecated the analyze API as all operations are now available on all endpoints. If you want to perform analysis (with expressions or functions) you can do so on either on the catalog or dataset endpoint. If you want to perform analysis on groups of records, you just have to group them with a `group_by` REST param.
```

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst (hunk 2 above)
Comment (benj-moreau) [reply source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:8]: +1 but IMO we don't need to say it is used internally in the doc.

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 5]
```diff
@@ -30,33 +25,31 @@ These APIs can be used, for example, to search for datasets and data, to compute
    * Each record provides values for each of the dataset's fields.
 
 
-ODS Search
-----------
+ODS Explore API
+---------------
 
-This ODS Search API is used internally by the Opendatasoft platform and widgets. To develop an application, we recommend using this API.
+The ODS Explore API is used internally by the Opendatasoft platform and widgets. To develop an application, we recommend using this API, in V2.
```
Comment (benj-moreau) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:31]: Maybe just : `is used in Opendatasoft widgets`

File: source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst [hunk 6]
```diff
@@ -28,15 +34,15 @@ These APIs can be used, for example, to search for datasets and data, to compute
...
+The ODS Explore API is used in Opendatasoft widgets. To develop an application, we recommend using this API, in V2.
 
 There are currently 2 versions of the ODS Explore API: the `V2 <https://help.opendatasoft.com/apis/ods-explore-v2/>`_ and the `V1 <https://help.opendatasoft.com/apis/ods-search-v1/>`_, which is now deprecated. Listed below, the main differences between the 2 versions.
 
 * Key changes in the V2:
 
   * A richer query language (ODSQL) is available, with new functions and arithmetic expressions.
   * Special filters (such as geofilter and facet filters) are now directly integrated to ODSQL, in the ``where`` parameter.
-  * Aggregations endpoint and records endpoint now share the same structure and ODSQL language.
+  * We deprecated the analyze API as all operations are now available on all endpoints. If you want to perform analysis (with expressions or functions) you can do so on either on the catalog or dataset endpoint. If you want to perform analysis on groups of records, you just have to group them with a `group_by` REST param.
```
Comment (baptabl) [inline source/exploring_catalog_and_datasets/04_getting_involved/using_api.rst:45]: Sorry, there was some typo in my sentence:
```
... you can do it either on the catalog or on the dataset endpoint.
```

**Pull request merged**